### PR TITLE
Earn battle points from trainer battles (with a variable)

### PR DESCRIPTION
**Pull request by laserXdolphin**
This allows the earning of Battle Points instead of money for scripted trainer battles by setting two variables.

1. setvar VAR_0x8003, 1 // enable battle points routine and battle frontier transitions if the config is set to true
2. setvar VAR_0x8004, 5 // the amount of BP that should be earned from this battle

## Description
Mainly implemented the tutorial Scyrous made with slight changes:

1. Used version 1.9.1 of the expansion (different BATTLESTRINGS_COUNT and Cmd_getmoneyreward).
2. Added config for enabling battle frontier transition
3. VAR_0x8003 and VAR_0x8004 are always reset in Cmd_getmoneyreward(void)

## **People who collaborated with me in this PR**
This was originally [written by](https://www.pokecommunity.com/showpost.php?p=10422051) by @Scyrous and can be found in the [pret scripting tutorial section](https://github.com/pret/pokeemerald/wiki/Tutorials#scripting).
[Original commit](https://github.com/Scyrous/pokeemerald/commit/f69452cb557bc2b06bf68fab67a6920621e28541)

## **Discord contact info**
`laserxdolphin`

### CODE REVIEW

File: src/battle_setup.c [hunk 1]
```diff
@@ -899,6 +899,11 @@ u8 GetTrainerBattleTransition(void)
     u32 trainerId = SanitizeTrainerId(gTrainerBattleOpponent_A);
     u32 trainerClass = GetTrainerClassFromId(gTrainerBattleOpponent_A);
 
+    #if BATTLE_FRONTIER_INTRO_FOR_BP_FIGHTS
+    if (VarGet(gSpecialVar_0x8003) == 1)
+        return GetSpecialBattleTransition(FRONTIER_MODE_DOUBLES);
+    #endif
```
Comment (AlexOn1ine) [inline src/battle_setup.c:905]: the preproc block should be a normal if statement

Comment (laserXdolphin) [reply src/battle_setup.c:905]: I'll gladly make the change but could you please tell me why that is the preferred way?

Comment (AlexOn1ine) [reply src/battle_setup.c]: because we can't verify if the code compiles with a preproc if it is set to false.

File: src/battle_setup.c [hunk 2]
```diff
@@ -899,6 +899,9 @@ u8 GetTrainerBattleTransition(void)
     u32 trainerId = SanitizeTrainerId(gTrainerBattleOpponent_A);
     u32 trainerClass = GetTrainerClassFromId(gTrainerBattleOpponent_A);
 
+    if (VarGet(gSpecialVar_0x8003) == 1 && BATTLE_FRONTIER_INTRO_FOR_BP_FIGHTS)
```
Comment (AlexOn1ine) [inline src/battle_setup.c:902]: can you switch those? The config should always be checked first

File: src/battle_setup.c (hunk 1 above)
Comment (laserXdolphin) [reply src/battle_setup.c]: That totally makes sense, thank you for the explanation.

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -7754,21 +7754,30 @@ static void Cmd_getmoneyreward(void)
...
+        }
     }
     else
     {
         s32 i, count;
         for (i = 0; i < PARTY_SIZE; i++)
         {
             if (GetMonData(&gPlayerParty[i], MON_DATA_SPECIES_OR_EGG) != SPECIES_NONE
-             && GetMonData(&gPlayerParty[i], MON_DATA_SPECIES_OR_EGG) != SPECIES_EGG)
+            && GetMonData(&gPlayerParty[i], MON_DATA_SPECIES_OR_EGG) != SPECIES_EGG)
```
Comment (AlexOn1ine) [inline src/battle_script_commands.c:7780]: the alignment here was correct. Please revert.

**Review by AlexOn1ine — COMMENTED** [review]
What's the easiest way to check Battle Points for review purposes?

File: src/battle_script_commands.c (hunk 3 above)
Comment (laserXdolphin) [reply src/battle_script_commands.c:7780]: I'd say the easiest way to review it, is to toggle the Frontier Pass in the Debug Menu under Flags&Vars.
review

**Comment by laserXdolphin** [comment]
I turned this into a draft because I found a bug where a trainer might drop bp.
I think it has something to do with repel reuse since gSpecialVar_0x8003 is set in sprays.c as well
`gSpecialVar_0x8003 = count;`
Possible solutions would be to use a number count can't reach for the activation condition or to use a different special var.

**Comment by AlexOn1ine** [comment]
> I turned this into a draft because I found a bug where a trainer might drop bp. I think it has something to do with repel reuse since gSpecialVar_0x8003 is set in sprays.c as well `gSpecialVar_0x8003 = count;` Possible solutions would be to use a number count can't reach for the activation condition or to use a different special var. I have not found a way yet to reproduce this 100% but will post it once I find it.

Using a different special variable would probably be best here. Just make sure to confirm that it does not clash with a different variable.

**Comment by ghoulslash** [comment]
This should not depend on scripting variables. Instead I would suggest adding a callnative or specia to set an ewram variable

**Comment by Scyrous** [comment]
> This should not depend on scripting variables. Instead I would suggest adding a callnative or specia to set an ewram variable

I concur. While I'm happy to see my commit is useful to some people, the implementation should be heavily scrutinized as it's not very good and quite limited in scope. That said, I have no desire to revisit the code myself anytime soon - I don't even remember what I did or why (thanks to a long hiatus).

**Comment by laserXdolphin** [comment]
Sounds good, sadly I currently don´t have the time to work on it.

**Pull request closed without merging**